app/assets/javascripts/models/beta/task.js

```diff
@@ -0,0 +1,69 @@
+import httpService from '../../services/httpService';
+import changeCase from 'change-object-case';
+
+export function newTask(projectId, storyId, task) {
+  let newTask = {
+    name: task,
+    done: false
+  }
+  return httpService
+    .post(`/projects/${projectId}/stories/${storyId}/tasks`, newTask)
+    .then(({ data }) => changeCase.camelKeys(data, { recursive: true, arrayRecursive: true }))
+    .catch(function (error) {
+      console.log(error);
+    });
+};
+
+export function updateTaskStatus(projectId, storyId, taskId, status) {
+  const taskStatus = changeCase.snakeKeys(status);
+  return httpService
+    .put(`/projects/${projectId}/stories/${storyId}/tasks/${taskId}/`, taskStatus)
+    .then(({ data }) => changeCase.camelKeys(data, { recursive: true, arrayRecursive: true }))
+    .catch(function (error) {
+      console.log(error);
+    });
+};
+
+export function excludeTask(projectId, storyId, taskId) {
+  return httpService
+    .delete(`/projects/${projectId}/stories/${storyId}/tasks/${taskId}`)
+    .then(({ data }) => changeCase.camelKeys(data, { recursive: true, arrayRecursive: true }))
+    .catch(function (error) {
+      console.log(error);
+    });
+};
+
+export const addTask = (story, task) => {
+  return {
+    ...story,
+    tasks: [
+      ...story.tasks,
+      task
+    ]
+  };
+};
+
+export const updateTask = (story, changedTask) => {
+  return {
+    ...story,
+    tasks: story.tasks.map((task) => {
+        if (task.id === changedTask.id) {
+          return {
+            ...task,
+            ...changedTask
+          };
+        };
+        return task
+      }
+    )
+  };
+};
+
+export const deleteTask = (taskId, story) => {
+  return {
+    ...story,
+    tasks: [
+      ...story.tasks.filter((task) => task.id !== taskId)
```

Comment:
You don't need to spread the array here, you can do just: 

```js
tasks: story.tasks.filter((task) => task.id !== taskId)
```

`filter` will always return a new array anyway.